addon/ext/route.js

```diff
@@ -0,0 +1,14 @@
+import Ember from 'ember';
+
+export function installOnRoute(route = Ember.Route) {
+  route.reopen({
```

Comment:
if you make these POJOs mixins, they will be detectable and also will not apply more then once.